my-app/src/components/Header.js

```diff
@@ -1,9 +1,26 @@
-function Header() {
+import '../asset/css/Header.css'
+import React from 'react';
+
+function Header({ onSearchChange }) { 
+    const handleSearchInputChange = (e) => {
+      onSearchChange(e.target.value); 
+    };
```

Comment:
Chỗ này nếu còn thời gian thì e có thể áp dụng useDebound custom hook vào trong phần search nhé Nam
https://dev.to/codebayu/react-custom-hooks-usedebounce-4il9